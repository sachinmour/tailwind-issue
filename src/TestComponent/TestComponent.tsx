import React from "react";
import clsx from "clsx";
import { TestComponentProps } from "./TestComponent.types";

const TestComponent: React.FC<TestComponentProps> = ({ theme }) => (
  <div
    data-testid="test-component"
    className={clsx(
      "border border-black border-solid p-4 w-96 text-center",
      `${theme === "secondary" ? "bg-black text-white" : "bg-white text-black"}`
    )}
  >
    <h1 className="text-3xl">I'm the test component</h1>
    <h2>Made with love by Harvey</h2>
  </div>
);

export default TestComponent;
