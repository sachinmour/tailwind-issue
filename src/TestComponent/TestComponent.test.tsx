/**
 * @jest-environment jsdom
 */
import React from "react";
import { render } from "@testing-library/react";
import TestComponent from "./TestComponent";
import { TestComponentProps } from "./TestComponent.types";
import "jest-styled-components";

describe("Test Component", () => {
  let props: TestComponentProps;

  beforeEach(() => {
    props = {
      theme: "primary",
    };
  });

  const renderComponent = () => render(<TestComponent {...props} />);

  it("should have primary className with default props", () => {
    const { getByTestId } = renderComponent();

    const testComponent = getByTestId("test-component");

    expect(testComponent).toMatchSnapshot();
  });

  it("should have secondary className with theme set as secondary", () => {
    props.theme = "secondary";
    const { getByTestId } = renderComponent();

    const testComponent = getByTestId("test-component");

    expect(testComponent).toMatchSnapshot();
  });
});
