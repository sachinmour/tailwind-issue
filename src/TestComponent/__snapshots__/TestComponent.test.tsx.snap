// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component should have primary className with default props 1`] = `
<div
  class="border border-black border-solid p-4 w-96 text-center bg-white text-black"
  data-testid="test-component"
>
  <h1
    class="text-3xl"
  >
    I'm the test component
  </h1>
  <h2>
    Made with love by Harvey
  </h2>
</div>
`;

exports[`Test Component should have secondary className with theme set as secondary 1`] = `
<div
  class="border border-black border-solid p-4 w-96 text-center bg-black text-white"
  data-testid="test-component"
>
  <h1
    class="text-3xl"
  >
    I'm the test component
  </h1>
  <h2>
    Made with love by Harvey
  </h2>
</div>
`;
